app/models/csv_data_file.rb

```diff
@@ -35,7 +36,8 @@ def skip?(_row)
 private
 
   def read_file
-    CSV.foreach(@csv_path, headers: true, encoding: 'ISO8859-1:utf-8') do |row|
+    csv = URI.open(@csv_uri).read
+    CSV.parse(csv, headers: true, encoding: 'ISO8859-1:utf-8').select do |row|
```

Comment:
Doing it this way reads and parses the entire file in memory. When reading the schools data file (~57Mb) it appeared to be using 1Gb of memory to do this (are PaaS instances are 2Gb mem?), which is why I went with the download-in-chunks and `CSV.foreach` route.

Reply:
I don't think the `encoding` param does anything with `CSV.parse`

Reply:
Good call - I've put this back to using `CSV.foreach` in combination with downloading any remote URI to a `Tempfile` (with your `RemoteFile.download`) if needed, and that seems to do the trick. I ran `rake import:schools` while monitoring RAM usage, and it never went over 185M (compared to 1.25G before)